File: backend/app/schemas/traceability.py
"""Traceability schemas for lot genealogy."""

from pydantic import BaseModel

from app.schemas.lot import LotResponse


class TraceabilityResponse(BaseModel):
    """
    Traceability response schema - matches Node/Express output exactly.

    Node/Express returns:
    {
        "central": {...lot fields...},
        "parents": [...parent lots...],
        "children": [...child lots...]
    }
    """

    central: LotResponse
    parents: list[LotResponse]
    children: list[LotResponse]

File: flow-viz-react/src/lib/api/traceability.ts
import { apiFetch } from './client';
import type { TraceabilityResponse } from './types';

/**
 * Get traceability graph for a lot
 * Returns central lot with parents and children
 */
export async function getTraceability(lotCode: string): Promise<TraceabilityResponse> {
  return apiFetch<TraceabilityResponse>(`/api/traceability/${encodeURIComponent(lotCode)}`);
}
